boot/bootutil/src/image_validate.c

```diff
@@ -488,7 +489,8 @@ bootutil_img_validate(struct enc_key_data *enc_state, int image_index,
 
             FIH_CALL(boot_nv_security_counter_get, fih_rc, image_index,
                                                            &security_cnt);
-            if (fih_not_eq(fih_rc, FIH_SUCCESS)) {
+            if (FIH_NOT_EQ(fih_rc, FIH_SUCCESS)) {
+                FIH_SET(fih_rc, FIH_FAILURE);
```

Comment:
Is it intended to prevent flipping bits in `fih_rc` and might turn it to `FIH_SUCCESS`?